core/domain/topic_services_test.py

```diff
@@ -2440,6 +2849,118 @@ def mock_get_current_time_in_millisecs() -> int:
+
+        story_exp_ids = (
+            topic_services.get_all_published_story_exploration_ids(
+                self.TOPIC_ID))
+
+        self.assertItemsEqual(story_exp_ids, topic_exp_ids)
+
+    def test_get_published_story_exploration_ids_from_published_chapters_when_serial_chapter_feature_enabled( # pylint: disable=line-too-long
+        self
+    ) -> None:
+        topic_exp_ids = ['exp_1', 'exp_2', 'exp_3', 'exp_4']
+        topic_published_chapters_exp_ids = topic_exp_ids[:2]
+        self._publish_story_chapters_with_explorations(
+            self.TOPIC_ID, self.story_id_1, topic_exp_ids,
+            chapter_exp_ids=topic_published_chapters_exp_ids)
+        with self.swap_to_always_return(
+                feature_flag_services, 'is_feature_flag_enabled', True):
+            topic_services.generate_topic_summary(self.TOPIC_ID)
+
+        story_exp_ids = (
+            topic_services.get_all_published_story_exploration_ids(
+                self.TOPIC_ID))
+
+        self.assertItemsEqual(
+            story_exp_ids, topic_published_chapters_exp_ids)
+
+    def test_get_published_story_exploration_ids_in_all_topics_when_topic_id_not_given( # pylint: disable=line-too-long
+        self
+    ) -> None:
+        topic_id_2 = topic_fetchers.get_new_topic_id()
+        story_id_4 = 'story_4'
+        self.save_new_topic(
+            topic_id_2, self.user_id, name='Name 2',
+            abbreviated_name='name2', url_fragment='name-two',
+            description='Description',
+            canonical_story_ids=[story_id_4], additional_story_ids=[],
+            uncategorized_skill_ids=[], subtopics=[], next_subtopic_id=1)
+        self.save_new_story(story_id_4, self.user_id, topic_id_2)
+
+        exp_ids = ['exp_1', 'exp_2', 'exp_3', 'exp_4']
+        self._publish_story_chapters_with_explorations(
+            self.TOPIC_ID, self.story_id_1, exp_ids[:1])
+        self._publish_story_chapters_with_explorations(
+            self.TOPIC_ID, self.story_id_3, exp_ids[1:3])
+        self._publish_story_chapters_with_explorations(
+            topic_id_2, story_id_4, exp_ids[3:])
+
+        with self.swap_to_always_return(
+                feature_flag_services, 'is_feature_flag_enabled', False):
+            topic_services.generate_topic_summary(self.TOPIC_ID)
+
+            story_exp_ids = (
+                topic_services.get_all_published_story_exploration_ids())
+
+            self.assertItemsEqual(story_exp_ids, exp_ids)
+
+        with self.swap_to_always_return(
+                feature_flag_services, 'is_feature_flag_enabled', True):
+            topic_services.generate_topic_summary(self.TOPIC_ID)
+
+            story_exp_ids = (
+                topic_services.get_all_published_story_exploration_ids())
+
+            self.assertItemsEqual(story_exp_ids, exp_ids)
+
+    def test_get_published_story_exploration_ids_when_exp_ids_not_found( # pylint: disable=line-too-long
```

Comment:
More like "when mapping not precomputed"?

Reply:
Done.